02-peer-protocol.md

```diff
@@ -560,6 +1038,302 @@ ones it has sent so it can use them should they be requested by
+      channel's funding output
+
+##### Rationale
+
+The channel funding output must be added by the *opener*, who pays its fees.
+
+#### The `tx_complete` Message
+
+Upon receipt of consecutive `tx_complete`s, the receiving node:
+  - if is the *accepter*:
+    - MUST fail the negotiation if:
+      - no funding output was received
+      - the value of the funding output is not equal to the sum of
+        `open_channel2`.`funding_satoshis` and `accept_channel2`.
+        `funding_satoshis`
+      - the value of the funding output is less than the `dust_limit`
+ - if it is an RBF attempt:
+    - MUST fail the negotiation if:
+      - the transaction's total fees is less than the last
+        successfully negotiated transaction's fees
+      - the transaction does not share a common input with all previous
+        funding transactions
+
+### The `commitment_signed` Message
+
+This message is exchanged by both peers. It contains the signatures for
+the first commitment transaction.
+
+Rationale and Requirements are the same as listed below,
+for [`commitment_signed`](#commiting-updates-so-far-commitment_signed) with the following additions.
+
+#### Requirements
+
+The sending node:
+  - MUST send zero HTLCs.
+
+The receiving node:
+  - if the message has one or more HTLCs:
+    - MUST fail the negotiation
+  - if it has not already transmitted its `commitment_signed`:
+    - MUST send `commitment_signed`
+  - Otherwise:
+    - MUST send `tx_signatures`
+
+#### Rationale
+
+The first commitment transaction has no HTLCs.
+
+### Sharing funding signatures: `tx_signatures`
+
+After a valid `commitment_signature` has been received
+from the peer and a `commitment_signature` has been sent, a peer:
+  - MUST transmit a [`tx_signatures` message](#the-tx_signatures-message) with their signatures for
+    the funding transaction
+
+#### Requirements
+
+The sending node:
+  - MUST verify it has received a valid commitment signature from its peer
+  - MUST remember the details of this funding transaction
+  - if it has NOT received a valid `commitment_signed` message:
+    - MUST NOT send a `tx_signatures` message
+
+The receiving node:
+  - if the `witness_stack` weight lowers the effective `feerate`
+    below the the *opener*'s feerate for the funding transaction:
+    - SHOULD broadcast their commitment transaction, closing the channel.
+  - SHOULD apply `witnesses` to the funding transaction and broadcast it
+  - if has already sent or received a `channel_ready` message for this
+    channel:
+    - MUST ignore this message
+
+#### Rationale
+
+A peer sends their `tx_signatures` as soon as they have received a valid
+`commitment_signed` message.
+
+The channel should be preemptively closed in the case where a peer provides
+valid witness data that causes their paid feerate to fall beneath the
+`open_channel2.funding_feerate_perkw` rate. This penalizes the peer
+for underpayment of fees.
```

Comment:
> Yeah, the 'double spending of an input to cancel an open' is a bit implicit here, it's probably worthwhile to call it out explicitly.

Reply:
This can be studied in a version 0.2 of the specification, see ongoing conversation on the mailing list on this subject: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-November/021142.html